sdks/python/apache_beam/transforms/window.py

```diff
@@ -369,7 +369,7 @@ def merge(self, merge_context):
     to_merge = []
     for w in sorted(merge_context.windows, key=lambda w: w.start):
       if to_merge:
-        if end > w.start:
+        if end > w.start:  # pylint: disable=used-before-assignment
```

Comment:
You are right, I think that would be cleaner. It should be set to Timestamp.MIN_TIMESTAMP would be the correct value I believe. I will update this and ping you again.